internal/settings/settings.go

```diff
@@ -29,6 +31,23 @@ func (o *Options) Validate() error {
 	if len(o.ModulePaths) != 0 && len(o.ExcludeModulePaths) != 0 {
 		return fmt.Errorf("at most one of `rootModulePaths` and `excludeModulePaths` could be set")
 	}
+
+	// TODO: abstract out this validation logic
+	// since it's being used in both the places
```

Comment:
I appreciate the comment but I expect that it won't be a big deal as we would eventually deprecate the CLI options so the validation logic won't be needed there anymore and the duplication in short term is fine.

Reply:
You can add a comment above the other (duplicate) validation function to say that it's deprecated, so that if someone is ever tempted to make changes there they are more careful.